crates/manifest/src/schema/v2.rs

```diff
@@ -404,4 +587,147 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_check_disjoint() {
+        for (a, b) in [
+            ("foo:bar@0.1.0", "foo:bar@0.2.0"),
+            ("foo:bar/baz@0.1.0", "foo:bar/baz@0.2.0"),
+            ("foo:bar/baz@0.1.0", "foo:bar/bub@0.1.0"),
+            ("foo:bar@0.1.0", "foo:bar/bub@0.2.0"),
+            ("foo:bar@1.0.0", "foo:bar@2.0.0"),
+            ("foo:bar@0.1.0", "foo:bar@1.0.0"),
+            ("foo:bar/baz", "foo:bar/bub"),
+            ("foo:bar/baz@0.1.0-alpha", "foo:bar/baz@0.1.0-beta"),
+        ] {
+            let a: DependencyPackageName = a.parse().unwrap();
+            let b: DependencyPackageName = b.parse().unwrap();
+            ComponentDependencies::check_disjoint(&a, &b).unwrap(); // TODO: should method on depednecny name
```

Comment:
Maybe do an `expect` with the two strings so it's easier to see which one has failed?  Or have the test return `Result` and just `?` that sucker